proto/provenance/exchange/v1/query.proto

```diff
@@ -0,0 +1,295 @@
+    option (google.api.http) = {
+      get: "/provenance/exchange/v1/orders/market/{market_id}"
+      additional_bindings: {get: "/provenance/exchange/v1/market/{market_id}/orders"}
+    };
+  }
+
+  // QueryGetOwnerOrders looks up the orders from the provided owner address.
+  rpc QueryGetOwnerOrders(QueryGetOwnerOrdersRequest) returns (QueryGetOwnerOrdersResponse) {
+    option (google.api.http).get = "/provenance/exchange/v1/orders/owner/{owner}";
+  }
+
+  // QueryGetAssetOrders looks up the orders for a specific asset denom.
+  rpc QueryGetAssetOrders(QueryGetAssetOrdersRequest) returns (QueryGetAssetOrdersResponse) {
+    option (google.api.http).get = "/provenance/exchange/v1/orders/asset/{asset}";
+  }
+
+  // QueryGetAllOrders gets all orders in the exchange module.
+  rpc QueryGetAllOrders(QueryGetAllOrdersRequest) returns (QueryGetAllOrdersResponse) {
+    option (google.api.http).get = "/provenance/exchange/v1/orders";
+  }
+
+  // QueryGetMarket returns all the information and details about a market.
+  rpc QueryGetMarket(QueryGetMarketRequest) returns (QueryGetMarketResponse) {
+    option (google.api.http).get = "/provenance/exchange/v1/market/{market_id}";
+  }
+
+  // QueryGetAllMarkets returns brief information about each market.
+  rpc QueryGetAllMarkets(QueryGetAllMarketsRequest) returns (QueryGetAllMarketsResponse) {
+    option (google.api.http).get = "/provenance/exchange/v1/markets";
+  }
+
+  // QueryParams returns the exchange module parameters.
+  rpc QueryParams(QueryParamsRequest) returns (QueryParamsResponse) {
+    option (google.api.http).get = "/provenance/exchange/v1/params";
+  }
+
+  // QueryValidateCreateMarket checks the provided MsgGovCreateMarketResponse and returns any errors it might have.
+  rpc QueryValidateCreateMarket(QueryValidateCreateMarketRequest) returns (QueryValidateCreateMarketResponse) {
+    option (google.api.http).get = "/provenance/exchange/v1/validate/create_market";
+  }
+
+  // QueryValidateMarket checks for any problems with a market's setup.
+  rpc QueryValidateMarket(QueryValidateMarketRequest) returns (QueryValidateMarketResponse) {
+    option (google.api.http) = {
+      get: "/provenance/exchange/v1/validate/market/{market_id}"
+      additional_bindings: {get: "/provenance/exchange/v1/market/{market_id}/validate"}
+    };
+  }
+
+  // QueryValidateManageFees checks the provided MsgGovManageFeesRequest and returns any errors that it might have.
+  rpc QueryValidateManageFees(QueryValidateManageFeesRequest) returns (QueryValidateManageFeesResponse) {
+    option (google.api.http).get = "/provenance/exchange/v1/validate/manage_fees";
+  }
+}
+
+// QueryOrderFeeCalcRequest is a request message for the QueryOrderFeeCalc endpoint.
+// Exactly one of ask_order or bid_order must be provided.
+message QueryOrderFeeCalcRequest {
+  // ask_order is the ask order to calculate the fees for.
+  AskOrder ask_order = 2;
```

Comment:
Would it be easier for this message to contain a Order message? That would enforce the one of these types.